app/controllers/courses_controller.rb

```diff
@@ -3,8 +3,11 @@ class CoursesController < ApplicationController
   after_action :verify_authorized, except: [:index, :show]
 
   def index
-    @taken_course_ids = current_user.try(:course_ids)
-    @planned_course_ids = @active_plan.try(:course_ids)
+    if user_signed_in?
+      @active_plan = Plan
+                     .includes(taken_courses: [:course]) # needed for taken course btns
```

Comment:
Line is too long. [87/80]<br>Place the . on the previous line, together with the method call receiver.